psetv2/utils_test.go

```diff
@@ -0,0 +1,289 @@
+		return "", fmt.Errorf("cannot fund address with faucet: %s", res)
+	}
+
+	respBody := map[string]string{}
+	if err := json.Unmarshal(data, &respBody); err != nil {
+		return "", err
+	}
+	return respBody["txId"], nil
+}
+
+func mint(address string, quantity int, name string, ticker string) (string, string, error) {
+	baseUrl, err := apiBaseUrl()
+	if err != nil {
+		return "", "", err
+	}
+
+	url := fmt.Sprintf("%s/mint", baseUrl)
+	payload := map[string]interface{}{
+		"address":  address,
+		"quantity": quantity,
+		"name":     name,
+		"ticker":   ticker,
+	}
+	body, _ := json.Marshal(payload)
+
+	resp, err := http.Post(url, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		return "", "", err
+	}
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", "", err
+	}
+
+	if res := string(data); len(res) <= 0 || strings.Contains(res, "sendtoaddress") {
+		return "", "", fmt.Errorf("cannot fund address with minted asset: %s", res)
+	}
+
+	respBody := map[string]interface{}{}
+	if err := json.Unmarshal(data, &respBody); err != nil {
+		return "", "", err
+	}
+	return respBody["txId"].(string), respBody["asset"].(string), nil
+}
+
+func apiBaseUrl() (string, error) {
+	u, ok := os.LookupEnv("API_URL")
+	if !ok {
+		return "", fmt.Errorf("API_URL environment variable is not set")
+	}
+	return u, nil
+}
+
+func unspents(address string) ([]map[string]interface{}, error) {
+	getUtxos := func(address string) ([]interface{}, error) {
+		baseUrl, err := apiBaseUrl()
+		if err != nil {
+			return nil, err
+		}
+		url := fmt.Sprintf("%s/address/%s/utxo", baseUrl, address)
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, err
+		}
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		var respBody interface{}
+		if err := json.Unmarshal(data, &respBody); err != nil {
+			return nil, err
+		}
+		return respBody.([]interface{}), nil
+	}
+
+	utxos := []map[string]interface{}{}
+	for len(utxos) <= 0 {
+		time.Sleep(1 * time.Second)
+		u, err := getUtxos(address)
+		if err != nil {
+			return nil, err
+		}
+		for _, unspent := range u {
+			utxo := unspent.(map[string]interface{})
+			utxos = append(utxos, utxo)
+		}
+	}
+
+	return utxos, nil
+}
+
+func b2h(buf []byte) string {
+	return hex.EncodeToString(buf)
+}
+
+func h2b(str string) []byte {
+	buf, _ := hex.DecodeString(str)
+	return buf
+}
+
+func fetchTx(txId string) (string, error) {
+	baseUrl, err := apiBaseUrl()
+	if err != nil {
+		return "", err
+	}
+	url := fmt.Sprintf("%s/tx/%s/hex", baseUrl, txId)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(data), nil
+}
+
+type signOpts struct {
+	pubkeyScript []byte
+	script       []byte
+}
+
+func signTransaction(
+	p *psetv2.Pset,
+	privKeys []*btcec.PrivateKey,
+	scripts [][]byte,
+	forWitness bool,
```

Comment:
Woudl specify already version here? To make it self evident for Upcoming taproot update